src/protocols/http/abortableRequest.js

```diff
@@ -0,0 +1,166 @@
+const CHARSETS = [
+  'ascii',
+  'base64',
+  'binary',
+  'hex',
+  'ucs2',
+  'ucs-2',
+  'utf16le',
+  'utf-16le',
+  'utf8',
+  'utf-8',
+  'latin1'
+];
+
+class AbortableRequest {
+  constructor (protocol, timeout, options) {
+    this._protocol = protocol;
+    this._timeout = timeout;
+    this._options = options;
+    this._request = null;
+
+    this._options.timeout = this._timeout;
```

Comment:
may options be undefined or null ?
shall we `constructor (protocol, timeout, options = {}) {}`
To avoid potential issue while dealing with nested data inside this object ? 
![error](https://user-images.githubusercontent.com/10114634/73738146-30c84180-4744-11ea-9922-fce900565f0d.png)